src/mem.rs

```diff
@@ -131,18 +132,19 @@ pub enum FlushDecompress {
     /// data has yet to be processed.
     Finish = ffi::MZ_FINISH as isize,
 
-    #[doc(hidden)] _Nonexhaustive,
+    #[doc(hidden)]
+    _Nonexhaustive,
 }
 
 /// The inner state for an error when decompressing
-#[derive(Debug, Default)]
+#[derive(Debug, Default, PartialEq)]
 struct DecompressErrorInner {
     needs_dictionary: Option<u32>,
 }
 
 /// Error returned when a decompression object finds that the input stream of
 /// bytes was not a valid input stream of bytes.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
```

Comment:
Does this need to pick up `PartialEq`? Equating errors can be a difficult thing to do unfortunately :(

Reply:
removed